Configurable table renderers

tobi-or-not-tobi opened the pull request:
closes #8704

tobi-or-not-tobi committed:
improved table config, dropping arrays except for fields

tobi-or-not-tobi committed:
Align table config with new model from dev

tobi-or-not-tobi committed:
align my company table config with new table structure

tobi-or-not-tobi committed:
fix config for xl

tobi-or-not-tobi committed:
introduce table options interface

tobi-or-not-tobi committed:
extend tests

tobi-or-not-tobi committed:
deep merge options

tobi-or-not-tobi committed:
latest

tobi-or-not-tobi committed:
Merge branch 'feature/my-company' into feature/GH-8704

tobi-or-not-tobi committed:
drop sort icon

tobi-or-not-tobi committed:
Use configurable table components

tobi-or-not-tobi committed:
Add property renderers to all lists

tobi-or-not-tobi committed:
Fix tests

tobi-or-not-tobi committed:
property renderer unit tests

tobi-or-not-tobi committed:
drop test component

tobi-or-not-tobi committed:
refactor cell component naming

tobi-or-not-tobi committed:
prettier

dunqan commented on feature-libs/my-account/organization/src/components/shared/organization-table/organization-table.module.ts:
```diff
@@ -0,0 +1,27 @@
+    UnitCellComponent,
+  ],
+})
+export class OrganizationCellModule {}
```
Should be `OrganizationTableModule`?

dunqan commented on projects/storefrontlib/src/shared/components/table/table-renderer.service.ts:
```diff
@@ -0,0 +1,113 @@
+@Injectable({
+  providedIn: 'root',
+})
+export class TableRendererService {
```
Would be good to add some description to the service and its role.

dunqan commented on projects/storefrontlib/src/shared/components/table/table.component.ts:
```diff
@@ -52,13 +59,18 @@ export class TableComponent {
   set dataset(dataset: Table) {
     this._dataset = dataset;
+    // consider decoupling type or structure to avoid recreation of table cell components
+    // test breakpoint specific cell components
```
do we want to keep those comments?

tobi-or-not-tobi committed:
rename module

tobi-or-not-tobi committed:
improve docs

tobi-or-not-tobi committed:
drop obsolete notes

tobi-or-not-tobi committed:
add global header and data cell component configuration

tobi-or-not-tobi committed:
drop obsolete generics

tobi-or-not-tobi committed:
drop obsolete generics

tobi-or-not-tobi committed:
renaming

tobi-or-not-tobi committed:
render default header and item data

tobi-or-not-tobi committed:
Feature/gh 8704 base component (#8724)

Introduce new list UX/UI, introduce generic list component

tobi-or-not-tobi committed:
Add highlight color

tobi-or-not-tobi committed:
Drop component providers for current services

tobi-or-not-tobi committed:
Introduce unique key per list service

tobi-or-not-tobi committed:
Add -1 for tabIndex on anchor links

tobi-or-not-tobi committed:
Finalize accessiblity and introduce trackby

tobi-or-not-tobi committed:
Merge branch 'feature/my-company' into feature/GH-8704

Pucek9 reviewed (changes requested):
- missing translation in users/details or edit [organization:user.orgUnit]
- I can't open address details (NullInjectorError: No provider for CurrentUnitAddressService!)
- in details pages, while open sub-page there is missing active class styling (grey or red background)

tobi-or-not-tobi committed:
Feature/gh 8704 table layout (#8771)

Optimised table layouts

tobi-or-not-tobi committed:
Merge branch 'feature/my-company' into feature/GH-8704

Pucek9 committed:
visible selected area for details sections
fix for unavailable `unit/addresses/:id` page
visible selected address for unit/addresses
fix missing translation (user.orgUnit)

tobi-or-not-tobi committed:
drop host binding for default table components

tobi-or-not-tobi commented:
closing as this has been added by #8924 directly
